test-suites/test-aave/uniswapAdapters.leverageTrading.spec.ts

```diff
@@ -0,0 +1,422 @@
+
+        const params = buildLeverageTradingParams([
+          { useATokenAsFrom: false, useATokenAsTo: false, toAsset: weth.address },
+        ]);
+
+        await expect(
+          pool
+            .connect(user)
+            .flashLoan(
+              leverageTrading.address,
+              [dai.address],
+              [amountDaiToSwap],
+              [1],
+              userAddress,
+              params,
+              0
+            )
+        )
+          .to.emit(leverageTrading, 'Swapped')
+          .withArgs(dai.address, weth.address, amountDaiToSwap, expectedWETHAmount);
+
+        const leverageWethBalance = await weth.balanceOf(leverageTrading.address);
+        const leverageDaiBalance = await dai.balanceOf(leverageTrading.address);
+        const userDaiStableDebtAmount = await daiStableDebtContract.balanceOf(userAddress);
+        const userAEthBalance = await aWETH.balanceOf(userAddress);
+
+        expect(leverageWethBalance).to.be.eq(Zero);
+        expect(leverageDaiBalance).to.be.eq(Zero);
+        expect(userDaiStableDebtAmountBefore.add(amountDaiToSwap)).to.be.eq(
+          userDaiStableDebtAmount
+        );
+        expect(userAEthBalanceBefore.add(expectedWETHAmount)).to.be.eq(userAEthBalance);
+      });
+
+      it('should correctly swap tokens(2 or more) and deposit', async () => {
+        const { users, pool, weth, aWETH, dai, oracle, leverageTrading, helpersContract } = testEnv;
+        const user = users[0].signer;
+        const userAddress = users[0].address;
+
+        const amountDaiToSwap = await convertToCurrencyDecimals(dai.address, '100');
+
+        const daiPrice = await oracle.getAssetPrice(dai.address);
+        const daiDecimals = await dai.decimals();
+        const expectedWETHAmount = amountDaiToSwap.mul(daiPrice).div(ten.pow(daiDecimals));
+
+        const daiStableDebtTokenAddress = (
+          await helpersContract.getReserveTokensAddresses(dai.address)
+        ).stableDebtTokenAddress;
+
+        const daiStableDebtContract = await getContract<StableDebtToken>(
+          eContractid.StableDebtToken,
+          daiStableDebtTokenAddress
+        );
+
+        const userDaiStableDebtAmountBefore = await daiStableDebtContract.balanceOf(userAddress);
+
+        const liquidityToSwap = expectedWETHAmount;
+        const userAEthBalanceBefore = await aWETH.balanceOf(userAddress);
+
+        await mockUniswapRouter.connect(user).setAmountToReturn(dai.address, liquidityToSwap);
+
+        const params = buildLeverageTradingParams([
+          {
+            useATokenAsFrom: false,
+            useATokenAsTo: false,
+            toAsset: weth.address,
+          },
+          {
+            useATokenAsFrom: false,
+            useATokenAsTo: false,
+            toAsset: weth.address,
+          },
+        ]);
+
+        await pool
+          .connect(user)
+          .flashLoan(
+            leverageTrading.address,
+            [dai.address, dai.address],
```

Comment:
why specify one token several times? why not different tokens?

Reply:
In the testEnv there are reserves for only 2 tokens(weth and dai).